xcodeproj/internal/top_level_targets.bzl

```diff
@@ -222,6 +237,13 @@ def process_top_level_target(
     is_swift = SwiftInfo in target
     swift_info = target[SwiftInfo] if is_swift else None
 
+    # DEBUG BEGIN
+    if AppleResourceInfo in target:
+        apple_resource_info = target[AppleResourceInfo]
+        build_settings["ASSETCATALOG_COMPILER_APPICON_NAME"] = "AppIcon"
```

Comment:
@brentleyjones This is the magic setting that sets the `App Icon Source` in Xcode 13.4.1. 

Here are some questions.

### App Icon Name

As you can see, I hardcoded the name, `AppIcon`. I am not sure the best way to identify the icon set to use based upon what is available in the `rules_apple` providers. It looks like I can reach into `AppleResourceInfo.asset_catalogs` and look for entries that have a `.appiconset` suffix. Then, I could just pick the one named `AppIcon` or the first one. This feels pretty fragile to me.

### Xcode Version

I think that this is a recent change to Xcode (i.e., post 12.x). Do we need to consider adding a target Xcode version to `xcodeproj_rule`?

Reply:
> It looks like I can reach into `AppleResourceInfo.asset_catalogs` and look for entries that have a `.appiconset` suffix. Then, I could just pick the one named `AppIcon` or the first one. This feels pretty fragile to me.

Instead, let's add `app_icons` to `XcodeProjAutomaticTargetProcessingInfo` (defaulting it to `app_icons` for bundle providing targets) and look at the `app_icons` attribute. That has validation that makes sure it's actually app icons (sticker packs use `.stickersiconset` for example and tvOS apps use `.brandassets`) and it enforces that a single directory is used: https://github.com/bazelbuild/rules_apple/blob/efbb206cacdde114893772d0cef7c2c868546a36/apple/internal/resource_actions/actool.bzl#L126-L131

> I think that this is a recent change to Xcode (i.e., post 12.x). Do we need to consider adding a target Xcode version to `xcodeproj_rule`?

No. `ASSETCATALOG_COMPILER_APPICON_NAME` has been around for a long time, so we are safe just setting it.